### DISCUSSION TITLE
blog/mnist

### DISCUSSION THREAD

**Discussion by giscus** [HarryWangATX/website#117] [General]
# blog/mnist

Have you ever wondered how a computer recognizes hand written digits? Read more to implement your own neural network for this task!

https://www.harrycodes.com/blog/mnist

**Comment by peterwake** [comment]
Hi Harry, thanks for the article - we got it working :)

We have been trying to change the first 2 layers to use the sigmoid function so have added 

```python
def sigmoid(x):
    return 1/(np.exp(-x)+1) 

def d_sigmoid(x):
    return (np.exp(-x))/((np.exp(-x)+1)**2)
```

And then used `sigmoid` instead of `tanh`, and `d_sigmoid` instead of `(1 - np.power((Ax, 2))` .. but it's giving us 37% accuracy. Any suggestions on what we are doing wrong? Thanks!

**Comment by HarryWangATX** [comment]
Hey! Thanks for checking out my blog!

I think what you are doing is correct. You could simplify the derivative of the sigmoid to be `sigmoid(x) * (1-sigmoid(x))` keep in mind the dimensionality of the matrix if you do this. Poor performance on the MNIST dataset could also be because sigmoid presents the problem of the vanishing gradient around values of 0 and 1. Try to also check the loss at each step, is it decreasing or staying around the same. If it isn't going down steadily, there could be mistakes made in backpropagation.
